BC fix for AOTIModelPackageLoader() constructor defaults

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #149082

The default value for `run_single_threaded` was wrongly specified in the .cpp file instead of the header, breaking C++-side instantiation of `AOTIModelPackageLoader` with no arguments. This PR fixes this and adds a test for the use case of running with `AOTIModelPackageLoader` instead of `AOTIModelContainerRunner` on the C++ side.

cc @desertfire @chenyang78 @penguinwu @yushangdi @benjaminglass1

jbschlosser committed:
BC fix for AOTIModelPackageLoader() constructor defaults

[ghstack-poisoned]

jbschlosser committed:
Update on "BC fix for AOTIModelPackageLoader() constructor defaults"


The default value for `run_single_threaded` was wrongly specified in the .cpp file instead of the header, breaking C++-side instantiation of `AOTIModelPackageLoader` with no arguments. This PR fixes this and adds a test for the use case of running with `AOTIModelPackageLoader` instead of `AOTIModelContainerRunner` on the C++ side.

cc desertfire chenyang78 penguinwu yushangdi benjaminglass1

[ghstack-poisoned]

desertfire commented on test/cpp/aoti_inference/test.cpp:
```diff
@@ -76,6 +76,31 @@ void test_aoti_script(const std::string& device) {
+  const auto& ref_output_tensors =
+      data_loader.attr(outputs_attr.c_str()).toTensorList().vec();
+
+  torch::inductor::AOTIModelPackageLoader runner(model_so_path);
```
I don't think AOTIModelPackageLoader takes a .so file.

jbschlosser replied:
right that's true.

```cpp
torch::inductor::AOTIModelPackageLoader(model_path)
```

so the test needs to serialize a model .pt2 package first then load it

jbschlosser committed:
Update on "BC fix for AOTIModelPackageLoader() constructor defaults"


The default value for `run_single_threaded` was wrongly specified in the .cpp file instead of the header, breaking C++-side instantiation of `AOTIModelPackageLoader` with no arguments. This PR fixes this and adds a test for the use case of running with `AOTIModelPackageLoader` instead of `AOTIModelContainerRunner` on the C++ side.

cc desertfire chenyang78 penguinwu yushangdi benjaminglass1

[ghstack-poisoned]

desertfire commented:
xref https://github.com/pytorch/pytorch/pull/148601

jbschlosser commented:
@pytorchbot merge

jbschlosser commented:
@pytorchbot merge -i

etaf commented:
Hi, @jbschlosser , @desertfire  should we cherry-pick this fix to release 2.7? Without this fix, the example code in https://github.com/pytorch/pytorch/blob/main/docs/source/torch.compiler_aot_inductor.rst is getting compilation error:
```
[ 33%] Building CXX object CMakeFiles/aoti_example.dir/inference.cpp.o
/data/xinanlin/aoti-example/inference.cpp: In function ‘int main()’:
/data/xinanlin/aoti-example/inference.cpp:10:63: error: no matching function for call to ‘torch::inductor::AOTIModelPackageLoader::AOTIModelPackageLoader(const char [10])’
   10 |     torch::inductor::AOTIModelPackageLoader loader("model.pt2");

```

jbschlosser commented:
@pytorchbot cherry-pick --onto release/2.7 -c regression

jbschlosser commented:
@etaf yes that's the plan :)
